internal/discover/hooks.go

```diff
@@ -25,54 +25,73 @@ import (
 var _ Discover = (*Hook)(nil)
 
 // Devices returns an empty list of devices for a Hook discoverer.
-func (h Hook) Devices() ([]Device, error) {
+func (h *Hook) Devices() ([]Device, error) {
 	return nil, nil
 }
 
 // Mounts returns an empty list of mounts for a Hook discoverer.
-func (h Hook) Mounts() ([]Mount, error) {
+func (h *Hook) Mounts() ([]Mount, error) {
 	return nil, nil
 }
 
 // Hooks allows the Hook type to also implement the Discoverer interface.
 // It returns a single hook
-func (h Hook) Hooks() ([]Hook, error) {
-	return []Hook{h}, nil
+func (h *Hook) Hooks() ([]Hook, error) {
+	if h == nil {
+		return nil, nil
+	}
+
+	return []Hook{*h}, nil
 }
 
-// CreateCreateSymlinkHook creates a hook which creates a symlink from link -> target.
-func CreateCreateSymlinkHook(nvidiaCDIHookPath string, links []string) Discover {
-	if len(links) == 0 {
-		return None{}
-	}
+type CDIHook struct {
+	nvidiaCDIHookPath string
+	disabledHooks     map[string]bool
+}
 
-	var args []string
-	for _, link := range links {
-		args = append(args, "--link", link)
-	}
-	return CreateNvidiaCDIHook(
-		nvidiaCDIHookPath,
-		"create-symlinks",
-		args...,
-	)
+type HookCreator interface {
+	Create(string, ...string) *Hook
+	DisableHook(string)
```

Comment:
I don't like having `DisableHook` as a function here. I know we discussed it, but I think constructing the proper may beforehand and not having this change state is more easy to reason about.

Reply:
you mean that the list of disabled hooks will be passed down during `NewHookCreator` and not having a method to allow adding more hooks after that?

Reply:
Yes. That is what I mean. We may need to move the contrstructor around in `lib.go` a bit, but I think this is a better long-term solution.

Reply:
Done
>  but I think constructing the proper may beforehand and not having this change state is more easy to reason about.

Will be addressed in a followup PR